src/safeds/data/tabular/containers/_table.py

```diff
@@ -628,12 +659,24 @@ def add_row(self, row: Row) -> Table:
             A new table with the added row at the end.
 
         """
-        if self._schema != row.schema:
+        int_columns = []
+        if self.number_of_rows == 0:
+            int_columns = list(filter(lambda name: isinstance(row[name], int | np.int64), row.column_names))
+            if self.number_of_columns == 0:
+                for column in row.column_names:
+                    self._data[column] = Column(column, [])
+                self._schema = Schema._from_pandas_dataframe(self._data)
```

Comment:
```suggestion
                self._schema = Schema._from_pandas_dataframe(self._data)
            elif self.column_names != row.column_names:
                raise SchemaMismatchError
```

Reply:
Why would you only check for column names if u can check the whole schema (as currently done)?

Reply:
Currently you only check the schema in tables which contain at least one row. If you have 0 rows you have to check weather the column names are equal:

```py
from safeds.data.tabular.containers import Table, Row

table = Table({"Column1": [], "Column2": []})

table = table.add_row(Row({"Column1": 1, "Column3": 2}))
```

Currently this will raise a `ValueError` but it should raise a `SchemaMismatchError`

The types of empty tables are not important (at least in the `Table.__eq__` method only the column names of empty tables are being compared and not the types) therefore we do not need to check the whole schema.

Reply:
Ah yes, I mistook the indent of the elif there to be one upfront